src/translator/translator.h

```diff
@@ -282,36 +285,62 @@ class TranslateService : public ModelServiceTask {
     auto devices = Config::getDevices(options_);
     numDevices_ = devices.size();
 
+    ThreadPool threadPool(numDevices_, numDevices_);
+    scorers_.resize(numDevices_);
+    graphs_.resize(numDevices_);
+    
     // preload models
-    std::vector<std::vector<io::Item>> model_items_;
     auto models = options->get<std::vector<std::string>>("models");
-    for(auto model : models) {
-      auto items = io::loadItems(model);
-      model_items_.push_back(std::move(items));
+    if(options_->get<bool>("model-mmap", false)) {
+      for(auto model : models) {
+        ABORT_IF(!io::isBin(model), "Non-binarized models cannot be mmapped");
+        LOG(info, "Loading model from {}", model);
+        model_mmaps_.push_back(mio::mmap_source(model));
+      }
+    }
+    else {
+      for(auto model : models) {
+        LOG(info, "Loading model from {}", model);
```

Comment:
We should check if additional logging doesn't break some regression tests, but I think it does not.

Reply:
whenever we run the regression tests we can verify this

Reply:
All regression tests pass with the following compilation options: `CC=/usr/bin/gcc-8 CXX=/usr/bin/g++-8 CUDAHOSTCXX=/usr/bin/g++8 cmake -DDETERMINISTIC=on -DUSE_SENTENCEPIECE=ON -DUSE_FBGEMM=on -DCOMPILE_CPU=on -DCOMPILE_TESTS=ON -DCOMPILE_EXAMPLES=ON -DCOMPILE_SERVER=ON -DCUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda-11.1 -DCOMPILE_MAXWELL=on -DCOMPILE_PASCAL=on ..`

Is anything missing there?

Reply:
thanks for running those roman. that is not building for iOS, but I think that should be fine. that at least shows us that the canonical codepath has no regressions